diff --git a/node_modules/react-native/React/CoreModules/RCTAlertController.m b/node_modules/react-native/React/CoreModules/RCTAlertController.m
index 8cd0bab..d2b11ae 100644
--- a/node_modules/react-native/React/CoreModules/RCTAlertController.m
+++ b/node_modules/react-native/React/CoreModules/RCTAlertController.m
@@ -35,6 +35,7 @@ - (void)show:(BOOL)animated completion:(void (^)(void))completion
 
 - (void)hide
 {
+  _alertWindow.hidden = YES;
   _alertWindow = nil;
 }
 
